pulsar-websocket/src/main/java/com/yahoo/pulsar/websocket/AbstractWebSocketHandler.java

```diff
@@ -63,28 +63,34 @@ public void onWebSocketConnect(Session session) {
         if (service.isAuthenticationEnabled()) {
             try {
                 authRole = service.getAuthenticationService().authenticateHttpRequest(request);
-                log.info("[{}] Authenticated WebSocket producer {} on topic {}", session.getRemoteAddress(), authRole,
+                log.info("[{}] Authenticated WebSocket client {} on topic {}", session.getRemoteAddress(), authRole,
                         topic);
 
             } catch (AuthenticationException e) {
-                log.warn("[{}] Failed to authenticated WebSocket producer {} on topic {}: {}",
+                log.warn("[{}] Failed to authenticated WebSocket client {} on topic {}: {}",
                         session.getRemoteAddress(), authRole, topic, e.getMessage());
                 close(WebSocketError.AuthenticationError);
                 return;
             }
         }
 
         if (service.isAuthorizationEnabled()) {
-            final String role = authRole;
-            isAuthorized(authRole).thenApply(isAuthorized -> {
-                if(!isAuthorized) {
-                    log.warn("[{}] WebSocket Client [{}] is not authorized on topic {}", session.getRemoteAddress(), role,
+            try {
+                if(!isAuthorized(authRole)) {
```

Comment:
actually idea behind keeping `isAuthorized` async to avoid blocking-thread on authorization check. So, can we keep it async and on future-complete we can call `createClient(session)` so, we can still keep it async and avoid creation of producer/consumer on authorization failure.